src/shared/sleep-config.c

```diff
@@ -141,6 +144,123 @@ int read_battery_capacity_percentage(void) {
         return battery_capacity;
 }
 
+static void get_system_battery_identifier_hash(uint8_t ret[static SHA256_DIGEST_SIZE]) {
+        _cleanup_free_ char *bat_manufacturer = NULL, *bat_model_name = NULL, *bat_serial_number = NULL;
+        sd_id128_t machine_id, product_id;
+        struct sha256_ctx hash;
+        int r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/manufacturer", &bat_manufacturer);
+        if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/manufacturer is unavailable: %m");
+        else if (r < 0)
+               log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/manufacturer: %m");
```

Comment:
The ENOENT should be ignored, but other errors shouldn't, because they are likely configuration mistakes or so that need to be fixed, so return here and in other places too

Reply:
the function type is void so is there any other way to handle this?
return will not be allowed....

Reply:
change it to int

Reply:
thanks. that worked.